### PR TITLE
TouchScreen Triggers added and MFI Controller Update (Stuff)

### PR DESCRIPTION
**Pull request by brand175**
Adds touchscreen triggers functionality on device and updated EmulatorViewController.mm with MFI Triggers added.

### CODE REVIEW

File: shell/apple/emulator-ios/emulator/EmulatorViewController.mm [hunk 1]
```diff
@@ -55,52 +109,77 @@ -(void)emuThread
...
     [super viewDidLoad];
-	
+
+#if !TARGET_OS_TV
 	self.controllerView = [[PadViewController alloc] initWithNibName:@"PadViewController" bundle:nil];
-    
-    self.context = [[EAGLContext alloc] initWithAPI:kEAGLRenderingAPIOpenGLES2];
+#endif
+	
+    self.context = [[EAGLContext alloc] initWithAPI:kEAGLRenderingAPIOpenGLES3];
```
Comment (skmp) [inline shell/apple/emulator-ios/emulator/EmulatorViewController.mm]: Why upgrade the context to GLES3? Do all devices support it?

File: shell/apple/emulator-ios/emulator/EmulatorViewController.mm [hunk 2]
```diff
@@ -55,52 +109,77 @@ -(void)emuThread
 		strcat(Args[2],P);
 	}
 
+	MakeCurrentThreadRealTime();
```
Comment (skmp) [inline shell/apple/emulator-ios/emulator/EmulatorViewController.mm]: Does this really make a difference?

File: shell/apple/emulator-ios/emulator/EmulatorViewController.mm [hunk 3]
```diff
@@ -156,135 +248,334 @@ - (void)update
...
+	glUniform2f(crtUniform_EmulatedImageSize, 320, 640);
+	float finalResWidth = view.drawableWidth;
+	float finalResHeight = view.drawableHeight;
+	glUniform2f(crtUniform_FinalRes, finalResWidth, finalResHeight);
+#endif
+//	kdebug_signpost_end(10, 0, 0, 0, 0);
 }
 
+/*** CRT Shader borrowed from BrainDX for Provenance app ***/
+- (void)setupCRTShader
```
Comment (skmp) [inline shell/apple/emulator-ios/emulator/EmulatorViewController.mm]: I think in the future we want to have a more generic support for shader effects in the core, and not in the shell code. Looks okay though.

**Comment by skmp** [comment]
@brand175 Can you leave some feedback on the comments? Thanks :)

**Comment by brand175** [comment]
I merged the EmulatorViewController.mm from Joe matt's build and added the function of the Triggers to the MFI controller. OpenGL 3.0 is only iPhone 5S and above.

**Comment by LoungeKatt** [comment]
Is this different than what was being hosted at https://github.com/reicast/reicast-emulator/tree/jm/xcode-port?

**Comment by skmp** [comment]
Ahh I see.

@brand175 can you please coordinate with the team working on `jm/xcode-port` before opening a PR in the future? As @LoungeKatt explained, #768 is the issue to discuss about these changes. Thanks!

@LoungeKatt why don't you open a WIP MR so pepole know that code is not a dead branch? We could revert the merge and then open a new MR. How does that sound?

**Comment by LoungeKatt** [comment]
That will work. I can lay these changes overtop it so it all blends together.

**Comment by LoungeKatt** [comment]
Before I go too crazy trying to blend this together. @brand175 How much testing have you done with these changes integrated? I ask because `extendedgamepad` is changed to `gamepad`, but that was something designated by the plugin used to connect to the controllers. It was not a locally chosen name. If it works, it may be that Moga is different than the rest (similar to Android)

**Comment by brand175** [comment]
@LoungeKatt Everything is working fine with the MFI controller and touchscreen the last time I tested this. I don’t have a Mac anymore so i’m SOL.

**Pull request merged**